crowbar_framework/app/controllers/installer_controller.rb

```diff
@@ -0,0 +1,154 @@
+#
+# Copyright 2015, SUSE LINUX GmbH
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+class InstallerController < ApplicationController
+  skip_before_filter :enforce_installer
+  before_filter :hide_navigation
+  after_filter :post_install
+
+  def index
+    @steps = all_steps
+  end
+
+  #
+  # Perform Crowbar Installation
+  #
+  # Provides the restful api call for
+  # /installer/status 	GET 	return done steps, error and success
+  # returns a hash with an indicator if the installation failed/succeeded and the steps that are done
+  def status
+    respond_to do |format|
+      format.json { render json: steps_done }
+      format.html { redirect_to installer_url }
+    end
+  end
+
+  #
+  # Set the status of an installation step
+  #
+  # Provides the restful api call for
+  # /installer/update 	POST 	set the status
+  # touch a file in /var/lib/crowbar named after the step
+  # not really needed yet, but maybe later
+  def update
+    step = params[:step]
+    return render_not_found unless all_steps.include? step.to_sym
+    write_file(crowbar_lib_dir, step)
+    respond_to do |format|
+      format.json { head :ok }
+      format.html { redirect_to installer_url }
+    end
+  end
+
+  #
+  # Perform Crowbar Installation
+  #
+  # Provides the restful api call for
+  # /installer/install 	POST 	triggers install-chef-suse.sh
+  # the bash Process will be spawned in the background and therefore has not a direct return value which we can use here
+  def install
+    installed_ok = Rails.root.join(".crowbar-installed-ok")
+    crowbar_dir = Rails.root.join("..")
+    if installed_ok.exist?
+      msg = "If you want to reinstall please remove /opt/dell/crowbar_framework/crowbar-installed-ok"
+      respond_to do |format|
+        format.json { render text: msg and return }
+        format.html { redirect_to installer_url and return }
```

Comment:
Style/AndOr: Use `&&` instead of `and`. (https://github.com/bbatsov/ruby-style-guide#no-and-or-or)